src/preact/compat.js

```diff
@@ -1,15 +1,73 @@
+import {options} from /*OK*/ 'preact';
+
+import * as mode from '#core/mode';
+
 /**
- * todo(kvchari): replace with official solution from preact
- * copying forwardRef implementation from https://gist.github.com/developit/974b5e4e582df8e954c5a7981603cd37
- * preact contains a bug where importing forwardRef imports a lot of extraneous code due to side-effects.
- * see issue for more information https://github.com/preactjs/preact/issues/3295
+ * Checks if our VNode type has a `forwardRef_` sigil, in which case it was created by our forwardRef implementation.
+ * If so, we move the VNode's ref to it's props, which will be passed to the Component as its props.
+ *
+ * See VNode type at https://github.com/preactjs/preact/blob/55ee4bc069f84cf5e3af8f2e2f338f4e74aca4d8/src/create-element.js#L58-L76
+ *
+ * @param {{
+ *   type: (string|function()),
+ *   ref: *,
+ *   props: !Object,
+ * }} vnode
  */
-import {options} from /*OK*/ 'preact';
+function newDiff(vnode) {
+  if (vnode['type']?.forwardRef_) {
+    vnode['props']['ref'] = vnode['ref'];
+    vnode['ref'] = null;
+  }
+  oldDiff?.(vnode);
+}
+
+// `__b` is the known mangled name exported by preact. This is called during
+// the diffing algorithm with the constructed VNode, before the vnode is
+// actually used to diff.
+const oldDiff = options['__b'];
+options['__b'] = newDiff;
+
+const REACT_FORWARD_SYMBOL = self.Symbol?.for?.('react.forward_ref') || 0xf47;
+
+/**
+ * Reimplements forwardRef without dragging in everything from preact/compat.
+ * See https://github.com/preactjs/preact/issues/3295
+ *
+ * @param {function(P):R} Component
+ * @return {function(P):R}
+ * @template P
+ * @template R
+ */
+export function forwardRef(Component) {
+  /**
+   * @param {!Object} props
+   * @return {R}
+   */
+  function Forward(props) {
+    const {ref, ...clone} = props;
+    return Component(clone, ref);
+  }
+
+  // Is faithful react support necessary? This file should only be used in Preact mode,
```

Comment:
Not sure the answer to this question - I'd tried different combinations of implementations for this and they all seemed to fail some amount of unit tests for random reasons, so I'd opted to keep the implementation as close as possible to the the "real" thing